scala-package/core/src/main/scala/ml/dmlc/mxnet/module/DataParallelExecutorGroup.scala

```diff
@@ -0,0 +1,695 @@
+/*
+ * Licensed to the Apache Software Foundation (ASF) under one or more
+ * contributor license agreements.  See the NOTICE file distributed with
+ * this work for additional information regarding copyright ownership.
+ * The ASF licenses this file to You under the Apache License, Version 2.0
+ * (the "License"); you may not use this file except in compliance with
+ * the License.  You may obtain a copy of the License at
+ *
+ *    http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package ml.dmlc.mxnet.module
+
+import ml.dmlc.mxnet.DType.DType
+import ml.dmlc.mxnet._
+import ml.dmlc.mxnet.module.DataParallelExecutorGroup.Builder
+import org.slf4j.{Logger, LoggerFactory}
+
+import scala.collection.mutable
+import scala.collection.mutable.ArrayBuffer
+
+object DataParallelExecutorGroup {
+  private val logger: Logger = LoggerFactory.getLogger(classOf[DataParallelExecutorGroup])
+  // Load a list of arrays into a list of arrays specified by slices
+  private def loadGeneralMulti(data: Seq[NDArray],
+                               targets: Seq[Array[((Int, Int), NDArray)]],
+                               majorAxis: Seq[Int]): Unit = {
+    for (((dSrc, dTargets), axis) <- data zip targets zip majorAxis) {
```

Comment:
Hmm... not quite understand 'have meaningful variable for `data zip targets zip majorAxis`'